cpp/tests/io/parquet_test.cpp

```diff
@@ -1068,4 +1072,87 @@ TEST_F(ParquetWriterStressTest, DeviceWriteLargeTableWithValids)
   expect_tables_equal(custom_tbl.tbl->view(), expected->view());
 }
 
+TEST_F(ParquetReaderTest, UserBounds)
+{
+  // trying to read more rows than there are should result in
+  // receiving the properly capped # of rows
+  {
+    srand(31337);
+    auto expected = create_random_fixed_table<int>(4, 4, false);
+
+    auto filepath = temp_env->get_temp_filepath("TooManyRows.parquet");
+    cudf_io::write_parquet_args args{cudf_io::sink_info{filepath}, *expected};
+    cudf_io::write_parquet(args);
+
+    // attempt to read more rows than there actually are
+    cudf_io::read_parquet_args read_args{cudf_io::source_info{filepath}};
+    read_args.num_rows = 16;
+    auto result        = cudf_io::read_parquet(read_args);
+
+    // we should only get back 4 rows
+    EXPECT_EQ(result.tbl->view().column(0).size(), 4);
+  }
+
+  // trying to read past the end of the # of actual rows should result
+  // in empty columns.
+  {
+    srand(31337);
+    auto expected = create_random_fixed_table<int>(4, 4, false);
+
+    auto filepath = temp_env->get_temp_filepath("PastBounds.parquet");
+    cudf_io::write_parquet_args args{cudf_io::sink_info{filepath}, *expected};
+    cudf_io::write_parquet(args);
+
+    // attempt to read more rows than there actually are
+    cudf_io::read_parquet_args read_args{cudf_io::source_info{filepath}};
+    read_args.skip_rows = 4;
+    auto result         = cudf_io::read_parquet(read_args);
+
+    // we should get empty columns back
+    EXPECT_EQ(result.tbl->view().num_columns(), 4);
+    EXPECT_EQ(result.tbl->view().column(0).size(), 0);
+  }
+
+  // trying to read 0 rows should result in reading the whole file
+  // TODO : this is somewhat incongruous with skipping past all the rows
```

Comment:
the bug seems to be older than this PR so I would suggest just keeping `num_rows` to the default value of -1 in this test, since it should not be passing as-is.

Reply:
This particular test doesn't have much meaning if we leave num_rows as -1.   Maybe change it so that it passes 0 for num_rows and then does an xfail when it sees that it actually got back 4 rows?

Reply:
That sounds fine